include/envoy/http/header_map.h

```diff
@@ -111,6 +111,12 @@ class HeaderString {
   void setInteger(uint64_t value);
 
   /**
+   * Set the value of the string to a static string reference.
+   * @param static_value MUST point to static data.
```

Comment:
It seems a bit close to the by value vs. by reference distinction. `setByValue()` vs. `setByReference()`? Seems a bit clunky, but that's one idea to riff on perhaps.